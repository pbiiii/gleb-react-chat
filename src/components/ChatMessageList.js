import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Typography from '@material-ui/core/Typography';
import ChatMessage from './ChatMessage/ChatMessage';
import ActiveUserType from '../types/ActiveUserType';
import MessageType from '../types/MessageType';

const styles = theme => ({
    messagesWrapper: {
        height: '100%',
        width: '100%',
        overflowX: 'hidden',
        overflowY: 'scroll',
        paddingBottom: theme.spacing.unit * 14,
    },
    noMessages: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
});

const ChatMessageList = ({ classes, messages, activeUser }) => {
    const messagesExists = messages && messages.length > 0;
    return (
        <div className={classNames(classes.messagesWrapper, !messagesExists && classes.noMessages)}>
            {messagesExists ? (
                messages.map(message => (
                    <ChatMessage key={message._id} activeUser={activeUser} {...message} />
                ))
            ) : (
                <Typography variant="h5">There is no messages yet</Typography>
            )}
        </div>
    );
};

ChatMessageList.propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    activeUser: PropTypes.shape(ActiveUserType).isRequired,
    messages: PropTypes.arrayOf(PropTypes.shape(MessageType)).isRequired,
};

export default withStyles(styles)(ChatMessageList);
