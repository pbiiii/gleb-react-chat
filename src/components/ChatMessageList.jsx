import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Typography from '@material-ui/core/Typography';
import ChatMessage from './ChatMessage/ChatMessage';

const styles = theme => ({
    messagesWrapper: {
        height: '100%',
        width: '100%',
        overflowX: 'hidden',
        overflowY: 'scroll',
        paddingBottom: theme.spacing.unit * 14,
    },
    noMessages: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
});

const ChatMessageList = ({ classes, messages, activeUser }) => {
    const messagesExists = messages && messages.length > 0;
    return (
        <div className={classNames(classes.messagesWrapper, !messagesExists && classes.noMessages)}>
            {messagesExists ? (
                messages.map(message => (
                    <ChatMessage key={message._id} activeUser={activeUser} {...message} />
                ))
            ) : (
                <Typography variant="h5">There is no messages yet</Typography>
            )}
        </div>
    );
};

export default withStyles(styles)(ChatMessageList);
