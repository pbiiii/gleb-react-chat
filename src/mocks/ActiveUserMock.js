import UserMock from './UserMock';

export default {
    ...UserMock,
    isMember: false,
    isCreator: true,
    isChatMember: true,
};
