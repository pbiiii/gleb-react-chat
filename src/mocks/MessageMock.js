import ChatMock from './ChatMock';
import UserMock from './UserMock';

export default {
    _id: 'random_id',
    chatId: ChatMock._id,
    content: 'Mock content',
    sender: { UserMock },
    createdAt: '2019-01-10 13:00',
};
