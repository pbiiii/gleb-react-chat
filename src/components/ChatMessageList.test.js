/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessageList from './ChatMessageList';
import ActiveUserMock from '../mocks/ActiveUserMock';
import MessageMock from '../mocks/MessageMock';

const mockProps = {
    activeUser: { ...ActiveUserMock },
    messages: [MessageMock],
};

jest.mock('./ChatMessage/ChatMessage', () => () => 'ChatMessage');

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessageList {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
    const tree = renderer
        .create(<ChatMessageList {...mockProps} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
