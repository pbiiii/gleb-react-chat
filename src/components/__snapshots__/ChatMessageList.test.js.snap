// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="ChatMessageList-messagesWrapper-3"
>
  ChatMessage
</div>
`;
